### PR TITLE
Add APK canary scanner

### PR DESCRIPTION
**Pull request by hydraxman**
## Description

- Add APK canary executor
- Remove resources folder from Jacoco coverage computation base. 
image

## Pull Request Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [x] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [ ] No

## How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

File: agent/src/main/resources/apk_canary/matrix-apk-canary-2.1.0.jar
Comment (zhou9584) [inline agent/src/main/resources/apk_canary/matrix-apk-canary-2.1.0.jar:1]: Why not use Gradle to manage the dependency.
image

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+            return getApkReportFromJsonReport(reportFile);
+        } catch (IOException | InterruptedException e) {
+            LOGGER.error("error in analyzeApk", e);
+        }
+        return null;
+    }
+
+    public static String getAPKAnalysisItemName(String apkName) {
+        return apkName.replace("-signed-aligned", "").replace(".apk", "");
```
Comment (zhou9584) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java:132]: What is the purpose of replacing '-signed-aligned' by ''

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+            String command = String.format("java -jar %s --config %s", canaryJar.getAbsoluteFile(), newConfigFile.getAbsolutePath());
+            LOGGER.info(command);
+            process = runtime.exec(command, null, workingDirFile);
+
+            try (InputStream inputStream = process.getInputStream();
+                 InputStream errorStream = process.getErrorStream();
+                 BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
```
Comment (zhou9584) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java:107]: 在当前线程读取 process 的输出可能会阻塞，可以参考CommandOutputReceiver 用新开的线程去接收输出

Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java:107]: 1. 这个阻塞应该是by design，貌似不会有什么实质问题，本来就在waitfor。
2. CommandOutputReceiver直接用有点问题，这个类的实现可能还可以优化。
    - 使用了GBK的charset
    - 作为一个工具类，他关闭了流，如果上游也在关闭流，不确定是否有问题。

Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java:107]: 综上，先不采用`CommandOutputReceiver `

Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java:107]: 综上，先不采用`CommandOutputReceiver `

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java (hunk 1 above)
Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java]: Cleaned. Good callout.

**Review by zhou9584 — APPROVED** [review]

**Review by TedaLIEz — APPROVED** [review]

**Pull request merged**